### ISSUE TITLE
Maping ignored in hooks

### ISSUE THREAD

**Issue by guillaumecherel** [mawww/kakoune#1618]
After remapping a key

    map global normal j p

I cannot use the new key as a key filter when creating a hook:

    hook global NormalKey p '%sh{touch a}'

works and 

    hook global NormalKey j '%sh{touch a}'

does not (no error, but the hook isn't triggered when I press j). It confused me, but I'm not sure what is the correct behaviour.

**Comment by danr** [comment]
Right, the hook looks at what command is executed, and withuot any remappings commands and keys coincide.

You can use the `RawKey` hook to get the actual key pressed, but it does not know in which mode it is (normal, insert, etc), so you would have to circumvent this by setting up and tearing down the hook on events like `NormalBegin` and `NormalEnd`.
